plugins/modules/s3_sync.py

```diff
@@ -271,7 +270,11 @@
 # You should have received a copy of the GNU General Public License
 # along with calculate_multipart_etag.  If not, see <http://www.gnu.org/licenses/>.
 
-DEFAULT_CHUNK_SIZE = 5 * 1024 * 1024
+if HAS_BOTO3:
```

Comment:
We explicitly require Boto3 and Botocore for this module.  The verification of this is handled by AnsibleAWSModule.

Rather than putting this here I'd suggest:
- Up at line 249 where we import botocore, also import boto3
- After the import (but inside the same try section) define 
```
DEFAULT_CHUNK_SIZE = boto3.s3.transfer.TransferConfig().multipart_chunksize
```
- Remove the DEFAULT_CHUNK_SIZE definition that was here.

This this reduces the patch down to a 3 line patch.  (+2 & -1)

Reply:
@tremble I've tried to implement the fix with your suggestions but I encounter some issue during the sanity and linting phase. In addition, I'm more prone to be verbose in the code and avoid possible static code analysis issue (possible undefined variables in the code) even if they are false positives.

Reply:
What's wrong with

```
try:
    import botocore
    import boto3
    DEFAULT_CHUNK_SIZE = boto3.s3.transfer.TransferConfig().multipart_chunksize
except ImportError:
    DEFAULT_CHUNK_SIZE = 5 * 1024 * 1024
    pass  # Handled by AnsibleAWSModule
```

Reply:
that's a good idea, I will implement it.